pkg/tests/apis/openapi_snapshots/API_CLIENT_GENERATION.md

```diff
@@ -0,0 +1,179 @@
+This folder contains a rendered OpenAPI file for each group/version. The “real” OpenAPI is generated by the running server, but the files here are used to build static frontend clients.
+
+## Generate RTK API Clients
+
+To show the steps to follow, we are going to work on adding an API client to create a new dashboard. Just adapt the following guide to your use case.
+
+### 1. Specify `group` and `version`
+First, check if the `group` and the `version` are already created in [openapi_test.go](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/openapi_test.go). 
+<br/> If you need to add a new block, you can check for the right `group` and `version` in the backend API call that you want to replicate in the frontend.
+
+```jsx
+{
+    Group:   "dashboard.grafana.app",
+	Version: "v0alpha1",
+}
+```
+### 2. Run the `TestIntegrationOpenAPIs` test
+Note that it will fail the first time you run it, and, in the second one, it will generate the corresponding OpenAPI spec. You will find it in [openapi_snapshots](https://github.com/grafana/grafana/blob/main/pkg/tests/apis/openapi_snapshots).
+<br/>
+<br/>
+
+> Note: You won't need to follow these two steps if the `group` you will work within is already in the `openapi_test.go` file.
+
+<br/>
+
+### 3. Create the API definition
+In the folder `../public/app/features/{your_group_name}/api/` you have to create the `baseAPI.ts` file for your group. This file should have the following content:
+
+```jsx
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+import { createBaseQuery } from 'app/api/createBaseQuery';
+import { getAPIBaseURL } from 'app/api/utils';
+
+export const BASE_URL = getAPIBaseURL('dashboard.grafana.app', 'v0alpha1');
+
+export const baseAPI = createApi({
+  reducerPath: 'dashboardAPI',
+  baseQuery: createBaseQuery({
+    baseURL: BASE_URL,
+  }),
+  endpoints: () => ({}),
+});
+```
+This is the API definition for the specific group you're working with, where `getAPIBaseURL` should have the proper `group` and `version` as parameters. The `reducePath` should also be modified to match `group + API`: `dashboard` will be `dashboardAPI`, `iam` will be `iamAPI` and so on. 
+
+### 4. Add the output information
+Open [generate-rtk-apis.ts](https://github.com/grafana/grafana/blob/main/scripts/generate-rtk-apis.ts) and add the following information:
+
+| Data | Descritpion |
```

Comment:
We should tighten the list of needed properties to just the ones that need to be modified. After reducing the list, I'd add a description with the purpose of that property. Even the official documentation does not have a lot of information 🙃

Reply:
the ones we definitely need are: `apiFile`, `schemaFile`, `apiImport`, `filterEndpoints`, `tag` and `hooks` (both these need to be `true`)

Reply:
`argSuffix`/`responseSuffix` we should probably not mention, and i'll try to remove it from anywhere that's using it in a followup PR

Reply:
> the ones we definitely need are: apiFile, schemaFile, apiImport, filterEndpoints, tag and hooks (both these need to be true)

Can you tell me what they do and why they must be set to true? I think it is a valuable info for the doc

Reply:
`hooks` is so it generates and exports the hooks
`tags` is to automatically attach tags to endpoints. this is needed for proper cache invalidation. see [here](https://redux-toolkit.js.org/rtk-query/usage/automated-refetching#:~:text=RTK%20Query%20uses,an%20active%20subscription.) in the docs

Reply:
actually i've just realised, we probably want `hooks: false` 😂 

we only want to export the hooks from the index file, not the generated one